### PR TITLE
Update the starter tutorial introduction

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull request adds an .rst file for the introduction to the starter tutorial to allow for additional starting tutorial pages to be added. It is the first in a series of pull requests to add a starter tutorial that expands beyond `first_figure.py`.

Addresses #770 
 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by willschlitzer** [comment]
image
I tried following what GMT used for setting up it's multi-page tutorial, but the link for the tutorial intro (tutorials/starter-tutorial.rst) is not showing up in the deployed environment. Any idea how to fix this?

**Comment by meghanrjones** [comment]
> image @GenericMappingTools/pygmt-admin I tried following what GMT used for setting up it's multi-page tutorial, but the link for the tutorial intro (tutorials/starter-tutorial.rst) is not showing up in the deployed environment. Any idea how to fix this?

You need to update `doc/conf.py` for your new folder (e.g., https://github.com/GenericMappingTools/pygmt/pull/1603/files#diff-e170e9a7d787c21095c6c11bb25f0f1ff0294a42a46d45ba6fb5ed794e457624). There will be merge conflicts between this and main if #1603 gets merged, so you may want to wait to resolve those before making more updates.

**Comment by willschlitzer** [comment]
> You need to update `doc/conf.py` for your new folder (e.g., https://github.com/GenericMappingTools/pygmt/pull/1603/files#diff-e170e9a7d787c21095c6c11bb25f0f1ff0294a42a46d45ba6fb5ed794e457624). There will be merge conflicts between this and main if #1603 gets merged, so you may want to wait to resolve those before making more updates.

Thanks @meghanrjones! I wait until #1603 is merged and then I'll get this sorted.

**Comment by willschlitzer** [comment]
image

I manually merged the main branch and update `doc/conf.py` to include the starter tutorials folder, but I'm still not displaying the link to `starter-tutorial.rst`. I'm obviously missing some key detail here, but I'm not able to figure out where I went wrong (cue @meghanrjones and @weiji14 pointing out a very easy fix).

To be clear, I'm planning on having this follow the model used on the GMT documentation, in that there is a link to the tutorial main page, which has links to individual pages/examples, otherwise the entire tutorial would be a big file and cumbersome page.

File: doc/conf.py
Comment (meghanrjones) [inline doc/conf.py:92]:
```suggestion
            "../examples/starter-tutorial",
```

File: doc/index.rst
Comment (meghanrjones) [inline doc/index.rst:23]:
```suggestion
    starter-tutorial/index.rst
```

File: examples/tutorials/starter-tutorial.rst
Comment (meghanrjones) [inline examples/tutorials/starter-tutorial.rst:21]:
```suggestion
```

**Review by meghanrjones — COMMENTED** [review]
Here are some steps that will make this show in the "Getting Started" section and not under the "User Guide/tutorials" section:

1. Commit suggested changes
2. From the base of the repository:
    ```
    mkdir examples/starter-tutorial
    git mv examples/tutorials/starter-tutorial.rst examples/starter-tutorial/README.txt
    git mv examples/tutorials/starter-tutorials/first_figure.py examples/starter-tutorial/first_figure.py
    rmdir examples/tutorials/starter-tutorials/
    ```
3. Add `        "../examples/starter-tutorial",` after L68 in `doc/conf.py` (in `example_dirs`)
4. In L72 in `doc/conf.py`, change:
     `     "gallery_dirs": ["gallery", "tutorials", "projections"],`
    to
     `    "gallery_dirs": ["gallery", "tutorials", "starter-tutorial", "projections"],` 
5. Add `doc/starter-tutorial/` to `.gitignore`
6. Add `rm -rf starter-tutorial` to L51 in `doc/Makefile`

I made these changes locally to try it out. If you want, I could commit and push them to this branch to save time (and revert the changes later if you do not like the structure).

**Comment by willschlitzer** [comment]
It worked! Thanks a ton for the help, @meghanrjones!

**Comment by willschlitzer** [comment]
I'm marking this PR ready for review. I'll have future pull requests that will add tutorial pages, but I don't want this to get so big as to be a mammoth review process.

File: examples/starter-tutorial/README.txt
Comment (michaelgrund) [inline examples/starter-tutorial/README.txt:17]:
```suggestion
This tutorial assumes that PyGMT has been successfully [installed](https://www.pygmt.org/dev/install.html). A quick
```

**Comment by meghanrjones** [comment]
The section "Note for experienced GMT users" is gone from the updated tutorial. I agree that the starter tutorial is not the most helpful place for that information, but I think it would be useful to retain the explanation somewhere for those coming from the GMT world. One option would be to add a new "FAQ" page with a subquestion "How does PyGMT differ from GMT?". I think we're getting to the point where there are some recurring questions on the Forum that could go into an FAQ anyways (e.g., some possible solutions for the "No module named PyGMT error").

**Comment by willschlitzer** [comment]
> The section "Note for experienced GMT users" is gone from the updated tutorial. I agree that the starter tutorial is not the most helpful place for that information, but I think it would be useful to retain the explanation somewhere for those coming from the GMT world. One option would be to add a new "FAQ" page with a subquestion "How does PyGMT differ from GMT?". I think we're getting to the point where there are some recurring questions on the Forum that could go into an FAQ anyways (e.g., some possible solutions for the "No module named PyGMT error").

Are you envisioning adding an FAQ in this pull request or a separate issue/PR?

**Comment by meghanrjones** [comment]
> > The section "Note for experienced GMT users" is gone from the updated tutorial. I agree that the starter tutorial is not the most helpful place for that information, but I think it would be useful to retain the explanation somewhere for those coming from the GMT world. One option would be to add a new "FAQ" page with a subquestion "How does PyGMT differ from GMT?". I think we're getting to the point where there are some recurring questions on the Forum that could go into an FAQ anyways (e.g., some possible solutions for the "No module named PyGMT error").
> 
> Are you envisioning adding an FAQ in this pull request or a separate issue/PR?

I think a separate PR would be good.

File: .gitignore
Comment (seisman) [inline .gitignore:31]: Do you think it's better to rename `starter-tutorial` to `get-started`? I think `get-started` is more commonly used.

File: examples/starter-tutorial/README.txt
Comment (seisman) [inline examples/starter-tutorial/README.txt:13]: We already have "About PyGMT" in the Overview page (https://www.pygmt.org/v0.5.0/overview.html). Do you think we can remove it from here?

Comment (willschlitzer) [reply examples/starter-tutorial/README.txt:13]: Should there still be any background info on PyGMT or should the tutorial index page just have links to the tutorials?

File: .gitignore
Comment (willschlitzer) [reply .gitignore:31]: I'm fine with either way. I'll change it unless anyone has any issues with it?

Comment (weiji14) [reply .gitignore:31]: The heading for the 'First figure' tutorial is called 'Getting started' :sweat_smile: How about `quickstart` or `first-map`/`first-figure`?

Other 'getting-started' pages for reference:
- [`xarray`](https://xarray.pydata.org/en/stable/getting-started-guide/index.html)
- [`geopandas`](https://geopandas.org/en/stable/getting_started.html)
- [`pandas`](https://pandas.pydata.org/docs/getting_started/index.html)
- etc

Comment (willschlitzer) [reply .gitignore:31]: I think both `quickstart` and `first-map` make it seem like it is a one-page guide to install PyGMT and make a quick figure. I'm envisioning this tutorial as a series of pages that create different figures and progressively builds upon previous concepts.

File: examples/starter-tutorial/README.txt
Comment (willschlitzer) [reply examples/starter-tutorial/README.txt]: I removed it in af13124

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:10]: In the Vercel preview, there's an odd empty code block here (https://pygmt-55t3spnwe-gmt.vercel.app/get-started/first_figure.html#sphx-glr-get-started-first-figure-py), in the spot where the import statement goes.

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:55]:
```suggestion
# In addition to the region, an argument needs to be passed to ``coast`` to tell
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:121]:
```suggestion
# While that the map's colors, projection, and size have been set, the region
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:123]:
```suggestion
# that is being displayed is not apparent. A frame can be added to
# annotate the latitude and longitude of the region.
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:148]: I would prefer a different additional exercise for this first example because the next DCW release will include ISO codes for regions such as the Canary Islands meaning that users should no longer need to use trial and error to get the region correct. Are there lakes or rivers in this area, where something like 'Plot lakes using ``lakes`` parameter, passing a color such as ``darkblue`` as the argument' could be used?

**Review by meghanrjones — COMMENTED** [review]
Nice work!

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py:10]: I want to still explain `import pygmt` rather than start the tutorial with it. Any idea how to remove the empty code block?

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:9]:
```suggestion
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:28]:
```suggestion

# sphinx_gallery_thumbnail_number = 4
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [reply examples/get-started/first_figure.py:10]: The latest two code suggestions should work for this.

**Comment by meghanrjones** [comment]
With the latest structure, you'll want to add `rm -rf get-started` to the `clean` target in `doc/Makefile`.

**Comment by willschlitzer** [comment]
> rm -rf get-started

It looks like it's already in there, unless I'm missing something?

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py]: Slight variation of this. I made the first exercise one for plotting using the ISO code, and then added an exercise to change the color of the lakes.

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:102]:
```suggestion
# projections are explained in the :doc:`projection </projections/index>` gallery. For this example,
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:103]:
```suggestion
# the Mercator projection is set using ``"M"``. The width of the figure will be
```
These changes are suggested for consistency with earlier in the tutorial, when the
arguments were formatted as code.

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:104]:
```suggestion
# 10 centimeters, as set by ``"10c"``. The map size can also be set in inches
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:105]:
```suggestion
# using "i" (e.g. a 5 inch wide Mercator projection would use ``"M5i"``).
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:148]:
```suggestion
#    as a Python string to the ``"region"`` parameter.
```
You need either two or three spaces at the start of the second line for the list to be formatted properly - please check the output after this suggestion to check if it worked.

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:152]:
```suggestion
# 3. Change the color of the lakes (using the ``"lakes"`` parameter) to "red."
```

File: examples/get-started/first_figure.py
Comment (meghanrjones) [inline examples/get-started/first_figure.py:158]: I think you also need to add spaces to start of all lines after the first one to get this formatted correctly.

**Review by meghanrjones — APPROVED** [review]
Looks nice, well done!

File: examples/get-started/first_figure.py
Comment (michaelgrund) [inline examples/get-started/first_figure.py:97]:
```suggestion
# This figure now has its colors set, but there is no projection or size
```

**Review by michaelgrund — APPROVED** [review]
Great work @willschlitzer!

File: examples/get-started/first_figure.py
Comment (seisman) [inline examples/get-started/first_figure.py:66]: > If you are using a Python script instead of a notebook, use ``fig.show(method="external)`` to display the figure.

After PR #529, `fig.show()` should automatically open the preview image using an external PDF viewer when running a Python script. Thus `fig.show(method="external)` is no longer required.

File: examples/get-started/first_figure.py
Comment (seisman) [inline examples/get-started/first_figure.py:18]: > If you use a Python script instead, there will be a difference in how the figure is shown that is explained in the first example.

Please see my comment about `fig.show(method="external)` below.

File: examples/get-started/first_figure.py
Comment (seisman) [inline examples/get-started/first_figure.py:127]: Why don't use `frame=True` here? I think `frame=True` means using default settings, but `frame="a"` means plotting the frame with **a**nnotations.

File: examples/get-started/first_figure.py
Comment (seisman) [inline examples/get-started/first_figure.py:148]: > DCW ISO code

I think there is no "DCW ISO". There is only "ISO country code".

File: examples/get-started/first_figure.py
Comment (seisman) [inline examples/get-started/first_figure.py:149]: You didn't add double quotes around other parameters.
```suggestion
#    as a Python string to the ``region`` parameter.
```

File: examples/get-started/first_figure.py
Comment (seisman) [inline examples/get-started/first_figure.py:153]:
```suggestion
# 3. Change the color of the lakes (using the ``lakes`` parameter) to "red".
```

File: examples/get-started/first_figure.py
Comment (seisman) [inline examples/get-started/first_figure.py:157]: > When the region is set without using a list full of integers or floating numbers, the argument needs to be passed as a Python string.

If I understand you correctly, you mean passing a string like `-20/20/-30/30` to `region`, right? I don't think it's clearly explained by this sentence because it doesn't say that the string should contain slash-separated numbers.

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py:127]: Per my incorrect testing at #1671, I was under the impression that `frame=True` wouldn't work any more. I'll change it back.

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py:157]: I'm actually referring to passing `d` or `g` when it comes to region. While I know that the literal GMT string can be passed for region, I don't want to make a mention of it in the first section of the tutorial. I think bringing up too many options early on is confusing, and using a list for something like this is probably more familiar to Python users instead of a string with numbers.

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py]: fixed in dfc996d

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py]: f912e75

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py]: Fixed in  2024b6c

File: examples/get-started/first_figure.py
Comment (seisman) [reply examples/get-started/first_figure.py:153]: I see what you meant. I also agree that a string like `-20/20/-30/30` is not Pythonic and should not be covered in this starter tutoria.

**Review by seisman — APPROVED** [review]
Nice work!

File: examples/get-started/first_figure.py
Comment (weiji14) [reply examples/get-started/first_figure.py:153]: Actually, you could pass in a list of strings like `["10W", "20E", "30S", "40N"]` right? Not necessarily all numbers.

File: examples/get-started/README.txt
Comment (seisman) [reply examples/get-started/README.txt:10]: It's ReST, not Markdown. So it should be 

```
`Jupyter <https://jupyter.org>`__
```

File: examples/get-started/README.txt
Comment (seisman) [reply examples/get-started/README.txt:9]: The following directive should work:
```
:doc:`install`
```

File: examples/get-started/first_figure.py
Comment (seisman) [reply examples/get-started/first_figure.py:153]: > Actually, you could pass in a list of strings like `["10W", "20E", "30S", "40N"]` right? Not necessarily all numbers.

Yes, but it's not commonly used, and better not to introduce too many options.

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py:76]: I think that adds too much complexity to this starter tutorial (I had to look up CMYK and HSV to make sure they were color formats). I would imagine any user checking out PyGMT for the first time will be fine with the names of colors, let alone the RGB or hex formats.

File: examples/get-started/first_figure.py
Comment (willschlitzer) [reply examples/get-started/first_figure.py:2]: @weiji14 I'm not familiar with using binder; could you make a code suggestion to add tt?

File: examples/get-started/README.txt
Comment (willschlitzer) [reply examples/get-started/README.txt]: How do I add the text that I want to be linked ("installed")?

File: examples/get-started/first_figure.py
Comment (seisman) [reply examples/get-started/first_figure.py:76]: I agree with @willschlitzer.

File: examples/get-started/README.txt
Comment (seisman) [reply examples/get-started/README.txt]:
```
:doc:`installed <install>`
```
or 
```
:doc:`installed </install>`
```
should work.

**Review by seisman — APPROVED** [review]

File: examples/get-started/first_figure.py
Comment (seisman) [reply examples/get-started/first_figure.py:2]: Perhaps it's better to add the binder link in a separate PR. Ideally, we should modify the template to automatically add the binder links for each example.

Comment (weiji14) [reply examples/get-started/first_figure.py:2]: Ok, sounds good to me.

**Review by weiji14 — APPROVED** [review]
Nice work, thanks @willschlitzer!

**Pull request merged**
